File: src/main/java/io/curiousoft/ijudi/ordermanagement/service/BaseModel.java
package io.curiousoft.ijudi.ordermanagement.service;

import org.springframework.data.annotation.Id;

import java.util.Date;

public class BaseModel {

    @Id
    private String id;
    private Date createdDate = new Date();

    public BaseModel() {
    }

    public BaseModel(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/model/BaseModel.kt
package io.curiousoft.izinga.commons.model

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import java.time.LocalDateTime
import java.util.*
import kotlin.collections.HashMap

open class BaseModel {

    @Id
    var id: String? = null

    @JsonProperty(value = "date")
    @CreatedDate
    var createdDate = Date()

    @LastModifiedDate
    var modifiedDate: Date? = null

    var tag: MutableMap<String, String> = mutableMapOf()

    constructor()
    constructor(id: String?) {
        this.id = id
    }
}